### PR TITLE
falling ball

### PR DESCRIPTION
**Pull request by peterjmartinson**
The ball is not the new thing here, though it implements a constant acceleration strategy. 

Current
--------

Each new shape is created completely on its own.  The dev needs to remember all styles and other variables when creating a new shape, to keep consistent.

Enhancement
---------------

Add a Standards.jsx file to keep track of global variables.

### CODE REVIEW

File: src/components/Canvas/FallingBall.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// @ts-check
```
Comment (belcurv) [inline src/components/Canvas/FallingBall.jsx:1]: Curious: does vim grok this? Do you get real time type checking as you code?

File: src/components/Canvas/FallingBall.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import { CanvasBuilder } from './CanvasBuilder'
+import { GRAVITY, BLUE } from './Standards'
```
Comment (belcurv) [inline src/components/Canvas/FallingBall.jsx:4]: I love the idea of standards or constants.

But I think we should have multiple files or exports of object enums. Like

```js
export const Physics = Object.freeze({
  GRAVITY: { x: 0, y: 0.1 }
})

export const Color = Object.freeze({
  // ...
})
```

Comment (belcurv) [reply src/components/Canvas/FallingBall.jsx:4]: And we should think carefully about what these constants "being" to. Like, should that be maintained in this `/components` folder? Or should they be more top-level?

File: src/components/Canvas/FallingBall.jsx (hunk 1 above)
Comment (peterjmartinson) [reply src/components/Canvas/FallingBall.jsx:1]: I think I'd need to install a plugin - Vim doesn't support type checking out of the box.  Honestly, I should just use VSC for debugging.

File: src/components/Canvas/FallingBall.jsx (hunk 2 above)
Comment (peterjmartinson) [reply src/components/Canvas/FallingBall.jsx:4]: OK - I need more education on these object enums.
And, yes, let's discuss where to hold things like this as well.

Comment (belcurv) [reply src/components/Canvas/FallingBall.jsx:4]: It's just a structural / design choice. We don't technically need `Object.freeze()` but it ensures we never reassign these object properties.

Something like:

```js
export const Color = Object.freeze({
  RED: 'rgb(212, 42, 32)',
  YELLOW: 'rgb(250, 194, 43)',
  BLUE: 'rgb(14, 99, 142)'
})
```

Is manifest in code like:

```js
import { Physics, Color } from './Standards'

const ball = {
    color: Color.BLUE,
    // ...
    update () {
      this.velocity.x += Physics.GRAVITY.x
      this.velocity.y += Physics.GRAVITY.y
      // ...
```

The enclosing objects provide a "namespace" for our enumerated constants.

Comment (belcurv) [reply src/components/Canvas/FallingBall.jsx:4]: Closing this thread. I added an issue to iterate on this later

https://github.com/peterjmartinson/huygens-site/issues/14

**Review by belcurv — APPROVED** [review]

**Pull request merged**